### PR TITLE
added new turning forces

### PR DESCRIPTION
**Pull request by Babymcnerd**
changed the process for adding turn forces to turn the wheels, and then add the force

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -29,6 +29,13 @@ fn main() {
         .add_systems(Update, cast_ray)
         .run();
 }
+#[derive(Component, Default, Reflect)]
+#[reflect(Component)]
+enum Location {
```
Comment (lukemcneil) [inline src/main.rs:34]: Call this `TireLocation`

File: src/car.rs [hunk 2]
```diff
@@ -104,22 +110,45 @@ fn calculate_tire_acceleration_and_braking_forces(
...
-        ev_add_force_to_car.send(AddForceToCar {
-            force: turning_torque,
-            point: torque_translation,
-        });
-    } else if keys.pressed(KeyCode::A) {
+    let tire_grip_strength = 1.0;
+    for (tire_transform, tire) in &tires {
+        let car_transform = car.single();
+        let mut torque_translation = car_transform.0.translation.clone();
+        torque_translation += car_transform.0.rotation.mul_vec3(Vec3::new(3.0, 0.0, 0.0));
```
Comment (lukemcneil) [inline src/car.rs]: remove

File: src/car.rs [hunk 3]
```diff
@@ -104,22 +110,45 @@ fn calculate_tire_acceleration_and_braking_forces(
...
-    torque_translation += car_transform.rotation.mul_vec3(Vec3::new(3.0, 0.0, 0.0));
-    if keys.pressed(KeyCode::D) {
-        ev_add_force_to_car.send(AddForceToCar {
-            force: turning_torque,
-            point: torque_translation,
-        });
-    } else if keys.pressed(KeyCode::A) {
+    let tire_grip_strength = 1.0;
+    for (tire_transform, tire) in &tires {
+        let car_transform = car.single();
```
Comment (lukemcneil) [inline src/car.rs]: do `let (car_transform, car_velocity) in &tires {`

File: src/car.rs [hunk 4]
```diff
@@ -104,22 +110,45 @@ fn calculate_tire_acceleration_and_braking_forces(
...
+        let tire_velocity = car_transform
+            .1
+            .linear_velocity_at_point(tire_transform.translation(), car_transform.0.translation);
+        let steering_velocity = steering_direction.dot(tire_velocity);
+        let desired_velocity_change = -steering_velocity * tire_grip_strength;
+        let desired_acceleration = desired_velocity_change;
+        println!(
+            "steering direction {}\ndesired acceleration {}",
+            steering_direction, desired_acceleration
+        );
```
Comment (lukemcneil) [inline src/car.rs]: can remove these I think

**Comment by Babymcnerd** [comment]
added new turning system

**Pull request merged**